packages/client/src/links/internals/httpUtils.ts

```diff
@@ -23,7 +23,7 @@ export interface HTTPLinkOptions {
    * Headers to be set on outgoing requests or a callback that of said headers
    * @link http://trpc.io/docs/v10/header
    */
-  headers?: HTTPHeaders | (() => HTTPHeaders | Promise<HTTPHeaders>);
+  headers?: HTTPHeaders | ((opts: {ops: Operation[]}) => HTTPHeaders | Promise<HTTPHeaders>);
```

Comment:
Here we reuse the `ops: Operation[]` for both batched and non regular http links. This should likely be changed once we have something working.

Reply:
A potentially nicer interface for the user was suggested by @KATT [here](https://github.com/trpc/trpc/issues/4071#issuecomment-1505130985) where only one operation is passed to HttpLink. 

If we want that I guess we would introduce two new types `HttpLinkOptions` and `HttpBatchLinkOptions`?

Reply:
After testing this out on our app, I actually like that we return a list of operations for both links. It means that I can share the same headers function between the http link and the batched one.

Reply:
I prefer to keep it separate, you can do a lower-level callback in both that is like `const getHeaders = (op: Operation) => { .... }`